exercises/handling-errors/solution/src/workflows.ts

```
@@ -0,0 +1,62 @@
+import { proxyActivities, ApplicationFailure, ActivityFailure, log, sleep } from '@temporalio/workflow';
+import type * as activities from './activities';
+import { Distance, OrderConfirmation, PizzaOrder} from './shared';
+
+const { sendBill, getDistance, validateCreditCard } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '5 seconds',
+});
+
+export async function pizzaWorkflow(order: PizzaOrder): Promise<OrderConfirmation> {
+  let totalPrice = 0;
+
+  // Validate the credit card number
+  try {
+    await validateCreditCard(order.customer.creditCardNumber);
+  } catch (err) {
+    if (err instanceof ActivityFailure && err.cause instanceof ApplicationFailure) {
```

You'll need to rethrow the error. The workflow won't fail unless you do so. And you certainly want to do that no matter if that was an `ApplicationFailure` or not.